go-controller/pkg/metrics/cluster_manager.go

```diff
@@ -0,0 +1,112 @@
+	Name:      "ready_duration_seconds",
+	Help:      "The duration for the cluster manager to get to ready state",
+})
+
+var metricV4HostSubnetCount = prometheus.NewGauge(prometheus.GaugeOpts{
+	Namespace: MetricOvnkubeNamespace,
+	Subsystem: MetricOvnkubeSubsystemClusterManager,
+	Name:      "num_v4_host_subnets",
+	Help:      "The total number of v4 host subnets possible",
+})
+
+var metricV6HostSubnetCount = prometheus.NewGauge(prometheus.GaugeOpts{
+	Namespace: MetricOvnkubeNamespace,
+	Subsystem: MetricOvnkubeSubsystemClusterManager,
+	Name:      "num_v6_host_subnets",
+	Help:      "The total number of v6 host subnets possible",
+})
+
+var metricV4AllocatedHostSubnetCount = prometheus.NewGauge(prometheus.GaugeOpts{
+	Namespace: MetricOvnkubeNamespace,
+	Subsystem: MetricOvnkubeSubsystemClusterManager,
+	Name:      "allocated_v4_host_subnets",
+	Help:      "The total number of v4 host subnets currently allocated",
+})
+
+var metricV6AllocatedHostSubnetCount = prometheus.NewGauge(prometheus.GaugeOpts{
+	Namespace: MetricOvnkubeNamespace,
+	Subsystem: MetricOvnkubeSubsystemClusterManager,
+	Name:      "allocated_v6_host_subnets",
+	Help:      "The total number of v6 host subnets currently allocated",
+})
+
+// RegisterClusterManagerBase registers ovnkube cluster manager base metrics with the Prometheus registry.
+// This function should only be called once.
+func RegisterClusterManagerBase() {
+	registerClusterManagerBaseMetrics.Do(func() {
+		prometheus.MustRegister(MetricClusterManagerLeader)
+		prometheus.MustRegister(MetricClusterManagerReadyDuration)
+		prometheus.MustRegister(prometheus.NewGaugeFunc(
+			prometheus.GaugeOpts{
+				Namespace: MetricOvnkubeNamespace,
+				Subsystem: MetricOvnkubeSubsystemClusterManager,
+				Name:      "build_info",
+				Help: "A metric with a constant '1' value labeled by version, revision, branch, " +
+					"and go version from which ovnkube was built and when and who built it",
+				ConstLabels: prometheus.Labels{
+					"version":    "0.0",
+					"revision":   config.Commit,
+					"branch":     config.Branch,
+					"build_user": config.BuildUser,
+					"build_date": config.BuildDate,
+					"goversion":  runtime.Version(),
+				},
+			},
+			func() float64 { return 1 },
+		))
+	})
+}
+
+// RegisterClusterManagerFunctional is a collection of metrics that help us understand ovnkube-cluster-manager functions. Call once after
+// LE is won.
+func RegisterClusterManagerFunctional() {
+	prometheus.MustRegister(metricV4HostSubnetCount)
+	prometheus.MustRegister(metricV6HostSubnetCount)
+	prometheus.MustRegister(metricV4AllocatedHostSubnetCount)
+	prometheus.MustRegister(metricV6AllocatedHostSubnetCount)
+}
+
+func UnregisterClusterManagerFunctional() {
+	prometheus.Unregister(metricV4HostSubnetCount)
+	prometheus.Unregister(metricV6HostSubnetCount)
+	prometheus.Unregister(metricV4AllocatedHostSubnetCount)
+	prometheus.Unregister(metricV6AllocatedHostSubnetCount)
+
+}
```

Comment:
Isn't "leader" and "ready_duration" missing from here ?

Reply:
Its defined in RegisterClusterManagerBase().

I think its the same for RegisterMasterBase().  I copied from there.